Packs/ProofpointThreatProtection/Integrations/ProofpointThreatProtection/ProofpointThreatProtection.py

```diff
@@ -0,0 +1,211 @@
+import demistomock as demisto  # noqa: F401
+from CommonServerPython import *  # noqa: F401
+
+
+''' CONSTANTS '''
+
+
+CTX_PREFIX = 'ProofpointThreatProtection'
+OSL_HEADER = 'Organizational Safe List'
+OBL_HEADER = 'Organizational Block List'
+URL_SUFFIX_SAFELIST = '/emailProtection/modules/spam/orgSafeList'
+URL_SUFFIX_BLOCKLIST = '/emailProtection/modules/spam/orgBlockList'
+AUTH_HOST_BASE_URL = 'https://auth.proofpoint.com/v1'
+
+
+''' INTEGRATION API CLIENT '''
+
+
+class Client(BaseClient):
+    """Client class to interact with the service API"""
+
+    def get_args(self):
+        return demisto.args()
+
+    def get_auth_host(self):
+        return AUTH_HOST_BASE_URL
+
+    def get_shared_integration_context(self):
+        return get_integration_context()
+
+    def set_shared_integration_context(self, context):
+        return set_integration_context(context)
+
+    def get_access_token(self, client_id, client_secret):
+        """
+        Get an access token that was previously created if it is still valid, else, generate a new access token from
+        the API key and secret.
+        """
+        # Check if there is an existing valid access token
+        integration_context = self.get_shared_integration_context()
+        if integration_context.get('access_token') and integration_context.get('expiry_time') > date_to_timestamp(datetime.now()):
+            return integration_context.get('access_token')
+        else:
+            try:
+                res = self._http_request(
+                    method='POST',
+                    full_url=f'{self.get_auth_host()}/token',
+                    headers={'Content-Type': 'application/x-www-form-urlencoded'},
+                    data={
+                        'grant_type': 'client_credentials',
+                        'client_id': client_id,
+                        'client_secret': client_secret
+                    }
+                )
+                if res.get('access_token'):
+                    expiry_time = date_to_timestamp(datetime.now(), date_format='%Y-%m-%dT%H:%M:%S')
+                    expiry_time += res.get('expires_in', 0) * 1000 - 10
```

Comment:
@thefrieddan1  I honestly can't speak to that one directly .... it sources from the original code base @randomizerxd  contributed .... but it would appear to be a direct code copy / carry over from the ServiceDeskPlus integration / content pack. Perhaps randty can shed some more light on it .... I remember asking myself the same question there ..... I mean I do understand the *1000 piece .... that essentially the difference between the 2 forms of Epoch timestamps found within XSOAR  code constantly .....  but as for the "- 10" .... I feel like I had arrived at answer but can not remember what that was .... @randomizerxd  thoughts?